runhouse/rns/secrets.py

```diff
@@ -117,14 +128,13 @@ def put(
         If from_env is True, will read secrets from environment variables instead of local config files.
         If file_path is provided, will read the secrets directly from the file
         If group is provided, will attribute the secrets to the specified group"""
-        provider = provider.lower()
-        if provider in cls.enabled_providers(as_str=True):
+        provider_name = provider.lower()
+        if provider_name in cls.builtin_providers(as_str=True):
             # if a supported cloud provider is given, use the provider's built-in class
-            provider_cls_name = cls.provider_cls_name(provider)
-            p = cls.get_class_from_name(provider_cls_name)
+            p = cls.builtin_provider_class_from_name(provider_name)
             if not from_env and not p.has_secrets_file():
                 # no secrets file configured for this provider
-                raise Exception(f"No local secrets file found for {provider}")
+                raise Exception(f"No local secrets file found for {provider_name}")
 
             secret = p.read_secrets(from_env=from_env, file_path=file_path)
```

Comment:
If the user passed in the secrets or file_path this is throwing an error for no reason (it doesn't matter that we didn't find one locally), or overwriting their provided secrets if we find others locally. We shouldn't enter this block if they passed in a secret, and we shouldn't throw an exception if they provided a file_path.